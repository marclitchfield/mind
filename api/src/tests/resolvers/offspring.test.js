import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Offspring tests', () => {
  applySetup();

  test('create offspring', () => {
    return verifyMutations('Offspring tests', 'create offspring', {
      mutations: [
        ({space}) => gql`mutation { Person { parent1:post_in_space(input:{sourceId:"${space.id}", title: "test parent"}) { id } } }`,
        ({space}) => gql`mutation { Person { parent2:post_in_space(input:{sourceId:"${space.id}", title: "test parent"}) { id } } }`,
        ({space}) => gql`mutation { Person { child:post_in_space(input:{sourceId:"${space.id}", title: "test child"}) { id } } }`,        
        ({space}) => gql`mutation { Event { event:post_in_space(input:{
          sourceId:"${space.id}", datetime: "2010-07-17" title: "test person"}) { id } }
        }`,
        ({parent1, parent2, child, event}) => gql`mutation { Offspring { post(input:{
          parent1:"${parent1.id}" parent2:"${parent2.id}" childId:"${child.id}" eventId:"${event.id}"}) { id } }
        }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            offspring {
              parents { title }
              child { title }
              event { title }
            }
            parents { title }
            children { title parents { title }}
          }
        }
      }`
    });
  });

  test('remove offspring', () => {
    return verifyMutations('Offspring tests', 'remove offspring', {
      mutations: [
        ({space}) => gql`mutation { Person { parent1:post_in_space(input:{sourceId:"${space.id}", title: "test parent"}) { id } } }`,
        ({space}) => gql`mutation { Person { parent2:post_in_space(input:{sourceId:"${space.id}", title: "test parent"}) { id } } }`,
        ({space}) => gql`mutation { Person { child:post_in_space(input:{sourceId:"${space.id}", title: "test child"}) { id } } }`,        
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", datetime: "2010-07-17" title: "test person"}) { id } } }`,
        ({parent1, parent2, child, event}) => gql`mutation { Offspring { offspring:post(input:{
          parent1:"${parent1.id}" parent2:"${parent2.id}" childId:"${child.id}" eventId:"${event.id}"}) { id } } 
        }`,
        ({parent1, parent2, child, event, offspring}) => gql`mutation { Offspring { post(input:{
          remove: true, id:"${offspring.id}" parent1:"${parent1.id}" parent2:"${parent2.id}" childId:"${child.id}" eventId:"${event.id}"}) { id } } 
        }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            offspring {
              parents { title }
              child { title }
              event { title }
            }
            parents { title }
            children { title parents { title }}
          }
        }
      }`
    });
  });
});