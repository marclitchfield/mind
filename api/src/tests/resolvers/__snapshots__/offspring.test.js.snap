// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offspring tests create offspring 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "children": Array [],
        "offspring": Array [],
        "parents": Array [
          Object {
            "__typename": "Person",
            "title": "test parent",
          },
          Object {
            "__typename": "Person",
            "title": "test parent",
          },
        ],
        "title": "test child",
      },
      Object {
        "__typename": "Person",
        "children": Array [
          Object {
            "__typename": "Person",
            "parents": Array [
              Object {
                "__typename": "Person",
                "title": "test parent",
              },
              Object {
                "__typename": "Person",
                "title": "test parent",
              },
            ],
            "title": "test child",
          },
        ],
        "offspring": Array [
          Object {
            "__typename": "Offspring",
            "child": Object {
              "__typename": "Person",
              "title": "test child",
            },
            "event": Object {
              "__typename": "Event",
              "title": "test person",
            },
            "parents": Array [],
          },
        ],
        "parents": Array [],
        "title": "test parent",
      },
      Object {
        "__typename": "Person",
        "children": Array [
          Object {
            "__typename": "Person",
            "parents": Array [
              Object {
                "__typename": "Person",
                "title": "test parent",
              },
              Object {
                "__typename": "Person",
                "title": "test parent",
              },
            ],
            "title": "test child",
          },
        ],
        "offspring": Array [
          Object {
            "__typename": "Offspring",
            "child": Object {
              "__typename": "Person",
              "title": "test child",
            },
            "event": Object {
              "__typename": "Event",
              "title": "test person",
            },
            "parents": Array [],
          },
        ],
        "parents": Array [],
        "title": "test parent",
      },
    ],
  },
}
`;

exports[`Offspring tests remove offspring 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "children": Array [],
        "offspring": Array [],
        "parents": Array [],
        "title": "test child",
      },
      Object {
        "__typename": "Person",
        "children": Array [],
        "offspring": Array [],
        "parents": Array [],
        "title": "test parent",
      },
      Object {
        "__typename": "Person",
        "children": Array [],
        "offspring": Array [],
        "parents": Array [],
        "title": "test parent",
      },
    ],
  },
}
`;
